explainers/security-privacy-questionnaire-post-task.md

```diff
@@ -0,0 +1,137 @@
+potential information leaks from timing-based side-channel attacks below.
+
+**Consideration 1**: Can delayed tasks be used to implement a high-resolution timer?
+
+This seems unlikely. `postTask()`'s delay, like `setTimeout()'s`, is expressed
+in whole milliseconds (the minimum non-zero delay being 1 ms), and there is no
+guarantee that tasks will run exactly when the delay expires since tasks are
+queued when the delay expires.
+
+**Consideration 2**:  Does `postTask()` leak information about other origins' tasks?
+
+The threat model we consider is code from two different origins running in
+separate event loops. Attackers could attempt to learn information about tasks
+running in another event loop by queuing `postTask()` tasks that are expected
+to run consecutively, e.g. two `'user-blocking'` tasks, and determining if they
+did. If not, then the UA may <sup>1</sup> have chosen to run another task in
+between.
+
+For such an attack to provide any information about tasks in another event
+loop, the event loops would need to be running in the same thread, since
+otherwise tasks would be running concurrently. But even then, there is no
+guarantee that the attacker could definitively determine that the delays are
+attributable to another event loop, e.g. *if* a task ran, it could be an
+internal browser task. And even if that could be determined, how tasks are
+chosen between event loops sharing a thread is not specified, so any
+information gained would be implementation-dependent.
+
+Our opinion is that any information gained in such an attack is likely to be
+benign. But if it is a concern, implementors can schedule between event loops
+in such a way that minimizes the risk, e.g. round-robin between event loops
+rather than use prioritization. Finally, we note that similar attacks could be
+carried out without this API, e.g. by scheduling tasks with `postMessage()`,
+although the implementation of `postTask()` priorities might change what
+implementation-dependent information could be obtained.
+
+<sup>1</sup>Another possibility is the UA chose to not run any tasks, e.g. it
+may have throttled the site.
+
+> 10. Do features in this specification enable new script execution/loading
+>     mechanisms?
+
+No. Note that this API requires a function, and does **not** support the
+`setTimeout()`-style string-to-script mechanism.
+
+> 11. Do features in this specification allow an origin to access other devices?
+
+No.
+
+> 12. Do features in this specification allow an origin some measure of control over
+>     a user agent's native UI?
+
+No.
+
+> 13. What temporary identifiers do the feautures in this specification create or
+>     expose to the web?
+
+None.
+
+> 14. How does this specification distinguish between behavior in first-party and
+>     third-party contexts?
+
+It does not make such a distinction. In the future, we may explore allowing
+first-party to assert some control over a third-party scheduling (e.g. imposing a maximum
```

Comment:
Do you want to explicitly distinguish capping an iframe's priority vs capping the priority of a script from a third-party origin running in a first-party context? There's tension between clearing up a common confusion, vs making the TAG think that you think we can enforce a security boundary between different scripts running in the same context.

Reply:
Sounds good, used your offline suggestion (thanks).